RecordHub.ChatService/RecordHub.ChatService.Api/Middlewares/GlobalExceptionHandlingMiddleware.cs

```diff
@@ -0,0 +1,52 @@
+﻿using RecordHub.Shared.DTO;
+using RecordHub.Shared.Exceptions;
+using System.Net;
+using System.Text.Json;
+
+namespace RecordHub.ChatService.Api.Middlewares
+{
+    public class GlobalExceptionHandlingMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger _logger;
+
+        public GlobalExceptionHandlingMiddleware(RequestDelegate next, ILogger<GlobalExceptionHandlingMiddleware> logger)
+        {
+            _next = next;
+            _logger = logger;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            try
+            {
+                await _next(context);
+            }
+            catch (EntityNotFoundException ex)
+            {
+                var code = HttpStatusCode.BadRequest;
+                _logger.LogWarning("{Exception}: {Message}, Source: {Source}",
```

Comment:
I think, it will be better to move log into the HandleExceptionAsync (pass there the exception). Moreover, you can replace base response dto with this exception and create dto inside the method. Or maybe i don't see any troubles, that can appear?